src/compiler/utilities.ts

```diff
@@ -6353,4 +6353,18 @@ namespace ts {
         }) as HeritageClause | undefined;
         return heritageClause?.token === SyntaxKind.ImplementsKeyword || heritageClause?.parent.kind === SyntaxKind.InterfaceDeclaration;
     }
+
+    export function isIdentifierTypeReference(node: Node): node is TypeReferenceNode & { typeName: Identifier } {
+        return isTypeReferenceNode(node) && isIdentifier(node.typeName);
+    }
+
+    export function arrayIsHomogeneous<T>(array: readonly T[], comparer: EqualityComparer<T> = equateValues) {

```

Comment:
I would really really like to recommend the obscure haskell name for this function but I can't remember it.

It's probably `eigenall`. See how much shorter that is?

Reply:
Serious comment: I wouldn't make the second parameter have a default until there's a reason for it, unless the default is really easy to get wrong.

Reply:
The issue is more that I really had to hunt for `equateValues`, which I think is the expected default for a function by this name. Using it as a default matches the convention set by `contains`, `arraysEqual`, `arrayIsEqualTo`, and `equalOwnProperties` in core.

Reply:
Fair enough; that's what I was getting at with "easy to get the default wrong".